pkg/bootstrap/boostrapkubeconfig.go

```diff
@@ -169,12 +169,21 @@ func GetKubeAPIServerAddress(ctx context.Context, client client.Client,
 		return klusterletConfig.Spec.HubKubeAPIServerURL, nil
 	}
 
+	if !helpers.DeployOnOCP() {
+		return "", fmt.Errorf("failed get Hub kube apiserver on non-OCP cluster")
```

Comment:
so here we return an error if it is not ocp, what function will be affected on non-ocp cluster?

Reply:
the bootstrap secret cannot be generated if cannot get hubkubeAPIServerURL here.

Reply:
so there will be no bootstrap secret on non-ocp cluster?

Reply:
bootstrap secret  is also needed in non-ocp cluster. And we only get hubkubeAPIServerURL from klusterletConfig in non-ocp cluster .